app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModel.kt

```diff
@@ -34,6 +37,20 @@ class TransactionDetailsViewModel
                 true == ownerCredentialsRepository.retrieveCredentials()?.let { credentials ->
             executionInfo.signers.contains(credentials.address) && !executionInfo.confirmations.map { it.signer }.contains(credentials.address)
         }
+
+    suspend fun validateSafeTxHash(transaction: TransactionDetails): Boolean {
+        kotlin.runCatching {
+            val safe = safeRepository.getActiveSafe()
+            val safeTxHash = (transaction.detailedExecutionInfo as DetailedExecutionInfo.MultisigExecutionDetails).safeTxHash
+            val calculatedSafeTxHash = calculateSafeTxHash(safe!!.address, transaction)?.toHexString()
+            safeTxHash == calculatedSafeTxHash
+        }.onSuccess {
+            return it
+        }.onFailure {
+            return false
+        }
+        return false
```

Comment:
instead of all these callbacks, I would suggest using `return Result{...}.getOrDefault(true)`